pkg/services/annotations/annotationsimpl/composite_store.go

```diff
@@ -0,0 +1,98 @@
+package annotationsimpl
+
+import (
+	"context"
+	"errors"
+	"slices"
+	"sync"
+
+	"github.com/grafana/grafana/pkg/services/annotations"
+	"github.com/grafana/grafana/pkg/services/annotations/accesscontrol"
+	"github.com/grafana/grafana/pkg/setting"
+)
+
+type CompositeStore struct {
+	primary   store
+	historian store
```

Comment:
Optionally: We could go a step further and build the `reader`/`writer` difference into the type system. This file could declare a `readOnlyStore` interface that's purely the read methods on the store.